vignettes/backtesting.Rmd

```diff
@@ -84,384 +74,363 @@ doctor_visits <- pub_covidcast(
-  rename(version = issue, percent_cli = value) %>%
+) |>
+  # The version date column is called `issue` in the Epidata API. Rename it.
+  rename(version = issue, percent_cli = value) |>
   as_epi_archive(compactify = TRUE)
 ```
 
-## Backtesting a simple autoregressive forecaster
-
-One of the most common use cases of `epiprocess::epi_archive()` object
-is for accurate model backtesting.
-
-In this section we will:
+In the interest of computational speed, we limit the dataset to 4 states and 
+2020-2021, but the full archive can be used in the same way and has performed
+well in the past.
 
-- develop a simple autoregressive forecaster that predicts the next value of the
-signal based on the current and past values of the signal itself, and
-- demonstrate how to slide this forecaster over the `epi_archive` object to
-produce forecasts at a few dates date, using version-unaware and -aware
-computations,
-- compare the two approaches.
+We choose this dataset in particular partly because it is revision heavy; for
+example, here is a plot that compares monthly snapshots of the data.
 
-To start, let's use a simple autoregressive forecaster to predict the percentage
-of doctor's hospital visits with CLI (COVID-like illness) (`percent_cli`) in the
-future (we choose this target because of the dataset's pattern of substantial
-revisions; forecasting doctor's visits is an unusual forecasting target
-otherwise). While some AR models output single point forecasts, we will use
-quantile regression to produce a point prediction along with an 90\% uncertainty
-band, represented by a predictive quantiles at the 5\% and 95\% levels (lower
-and upper endpoints of the uncertainty band).
+<details>
+<summary>Code for plotting</summary>
+```{r plot_revision_example, warn = FALSE}
+geo_choose <- "ca"
+forecast_dates <- seq(from = as.Date("2020-08-01"), to = as.Date("2021-11-01"), by = "1 month")
+percent_cli_data <- bind_rows(
+  # Snapshotted data for the version-faithful forecasts
+  map(
+    forecast_dates,
+    ~ doctor_visits |>
+      epix_as_of(.x) |>
+      mutate(version = .x)
+  ) |>
+    bind_rows() |>
+    mutate(version_faithful = "Version faithful"),
+  # Latest data for the version-un-faithful forecasts
+  doctor_visits |>
+    epix_as_of(doctor_visits$versions_end) |>
+    mutate(version_faithful = "Version un-faithful")
+)
 
-The `arx_forecaster()` function wraps the autoregressive forecaster we need and
-comes with sensible defaults:
+p0 <-
+  ggplot(data = percent_cli_data |> filter(geo_value == geo_choose)) +
+  geom_vline(aes(color = factor(version), xintercept = version), lty = 2) +
+  geom_line(
+    aes(x = time_value, y = percent_cli, color = factor(version)),
+    inherit.aes = FALSE, na.rm = TRUE
+  ) +
+  scale_x_date(breaks = "2 months", date_labels = "%b %Y") +
+  scale_y_continuous(expand = expansion(c(0, 0.05))) +
+  labs(x = "Date", y = "smoothed, day of week adjusted covid-like doctors visits") +
+  theme(legend.position = "none")
```

Comment:
You can probably replace some of this with something like the following:

```r
autoplot(
  archive_cases_dv_subset, percent_cli, 
  .versions = seq(ymd("2020-07-01"), ymd("2021-11-30"), by = "1 month"), 
  .mark_versions = TRUE
) +
  scale_x_date(minor_breaks = "month", date_labels = "%b %Y") +
  labs(x = "", y = "% of doctor's visits with\n Covid-like illness") + 
  scale_color_viridis_c(option = "B", end = .8) +
  scale_y_continuous(limits = c(0, NA), expand = expansion(c(0, 0.05))) +
  theme(legend.position = "none")
```

(I detest the rainbow colours.)

Reply:
Right, this was written before we'd merged that and I forgot to update it.

Incidentally, if we're using inferno, we should probably invert the color scale, since the latest version is black. Including both ^ and that flip in the next commit.